allennlp/tests/commands/dry_run_test.py

```diff
@@ -36,20 +41,121 @@ def setUp(self):
+
+        with open(extended_vocab_path / 'tokens.txt') as f:
+            tokens = [line.strip() for line in f]
+
+        assert tokens[0] == '@@UNKNOWN@@'
+        assert tokens[1] == 'some_weird_token_1'
+        assert tokens[2] == 'some_weird_token_2'
+
+        tokens.sort()
+        assert tokens == ['.', '@@UNKNOWN@@', 'animals', 'are',
+                          'some_weird_token_1', 'some_weird_token_2']
+
+        with open(extended_vocab_path / 'labels.txt') as f:
+            labels = [line.strip() for line in f]
+
+        labels.sort()
+        assert labels == ['N', 'V']
+
+    def test_dry_run_without_extension(self):
+        existing_serialization_dir = self.TEST_DIR / 'existing'
+        extended_serialization_dir = self.TEST_DIR / 'extended'
+        existing_vocab_path = existing_serialization_dir / 'vocabulary'
+        extended_vocab_path = extended_serialization_dir / 'vocabulary'
+
+        vocab = Vocabulary()
+        # if extend is False, its users responsibility to make sure that dataset instances
+        # will be indexible by provided vocabulary. At least @@UNKNOWN@@ should be present in
+        # namespace for which there could be OOV entries seen in dataset during indexing.
+        # For `tokens` ns, new words will be seen but `tokens` has @@UNKNOWN@@ token.
+        # but for 'labels' ns, there is no @@UNKNOWN@@ so required to add 'N', 'V' upfront.
+        vocab.add_token_to_namespace('some_weird_token_1', namespace='tokens')
+        vocab.add_token_to_namespace('some_weird_token_2', namespace='tokens')
+        vocab.add_token_to_namespace('N', namespace='labels')
+        vocab.add_token_to_namespace('V', namespace='labels')
+        os.makedirs(existing_serialization_dir, exist_ok=True)
+        vocab.save_to_files(existing_vocab_path)
+
+        self.params['vocabulary'] = {}
+        self.params['vocabulary']['directory_path'] = existing_vocab_path
+        self.params['vocabulary']['extend'] = False
+        dry_run_from_params(self.params, extended_serialization_dir)
+
+        with open(extended_vocab_path / 'tokens.txt') as f:
+            tokens = [line.strip() for line in f]
+
+        assert tokens[0] == '@@UNKNOWN@@'
+        assert tokens[1] == 'some_weird_token_1'
+        assert tokens[2] == 'some_weird_token_2'
+        assert len(tokens) == 3
+
+    def test_make_vocab_args(self):
+        parser = argparse.ArgumentParser(description="Testing")
+        subparsers = parser.add_subparsers(title='Commands', metavar='')
+        DryRun().add_subparser('dry-run', subparsers)
+        for serialization_arg in ["-s", "--serialization-dir"]:
+            raw_args = ["dry-run", "path/to/params", serialization_arg, "serialization_dir"]
+            args = parser.parse_args(raw_args)
+            assert args.func == dry_run_from_args
+            assert args.param_path == "path/to/params"
+            assert args.serialization_dir == "serialization_dir"
+
```

Comment:
@DeNeutoy  Do you think following two tests should be added? I was just planning to test if following two strings are logged on stdout "--Dataset Statistics--" and "--Vocabulary Statistics--" respectively. I tried very shortly something as suggested [here](https://stackoverflow.com/questions/20507601/writing-a-pytest-function-for-checking-the-output-on-console-stdout-in-python) but couldn't get it to work. Such hardcoded string checking doesn't seem correct in test, but don't know what else to test as far as print_statistics methods are concerned.

Btw, Running `pytest -s allennlp/tests/commands/dry_run_test.py` does show me that they are indeed printed.

Reply:
You don't need to test that things are printed, in this case - someone would notice, and it has no side-effects or experimental impact, so I think it's fine.